File: digitalpassbook2/app/src/main/java/com/example/digitalpassbook/Event.java
package com.example.digitalpassbook;
import com.google.gson.annotations.SerializedName;

public class Event {

    @SerializedName("id")
    private int id;

    @SerializedName("orgId")
    private int orgId;

    @SerializedName("name")
    private String name;

    @SerializedName("description")
    private String description;

    @SerializedName("date")
    private String date;

    @SerializedName("start")
    private String start;

    @SerializedName("end")
    private String end;

    @SerializedName("location")
    private String location;

    public Event (int orgID, String eventName, String description, String date,
                  String starttime, String endtime, String location) {
        this.orgId = orgID;
        this.name = eventName;
        this.description = description;
        this.date = date;
        this.start = starttime;
        this.end = endtime;
        this.location = location;
    }
    public Event (int orgID, String eventName) {
        this.orgId = orgID;
        this.name = eventName;
        this.description = " ";
        this.date = " ";
        this.start = " ";
        this.end = " ";
        this.location = " ";
    }

    public int getId() { return this.id; }
    public int getOrgId() { return this.orgId; }
    public String getName() { return this.name; }
    public String getDescription() { return this.description; }
    public String getDate() { return this.date; }
    public String getStartTime() { return this.start; }
    public String getEndTime() { return this.end; }
    public String getLocation() { return this.location; }
}

File: digitalpassbook2/app/src/main/java/com/example/digitalpassbook2/server/Event.kt
package com.example.digitalpassbook2.server

import com.google.gson.annotations.SerializedName
import java.util.*

class Event {
    @SerializedName("id")
    val id = 0

    @SerializedName("orgId")
    var orgId: Int

    @SerializedName("date")
    var startDate: String

    @SerializedName("endDate")
    var endDate: String

    @SerializedName("eventName")
    var name: String

    @SerializedName("description")
    var description: String

    @SerializedName("location")
    var location: String

    @SerializedName("transferability")
    var transferability: Boolean

    @SerializedName("openTimeVisibility")
    var openTimeVisibility: Boolean

    @SerializedName("closeDateVisibility")
    var closeDateVisibility: Boolean

    @SerializedName("closeTimeVisibility")
    var closeTimeVisibility: Boolean

    @SerializedName("cutoff")
    var cutoff: String

    @SerializedName("allStudentsVisibility")
    var allStudentsVisibility: Boolean

    @SerializedName("bouncers")
    var bouncers: Array<String>

    @SerializedName("numArrived")
    var numArrived: Int

    constructor(orgID: Int, startDate: String, endDate: String, eventName: String, description: String,
                location: String, transferability: Boolean, openTimeVisibility: Boolean,
                closeDateVisibility: Boolean, closeTimeVisibility: Boolean, cutoff : String,
                allStudentsVisibility: Boolean, bouncers: Array<String>, numArrived: Int) {
        this.orgId = orgID
        this.startDate = startDate
        this.endDate = endDate
        this.name = eventName
        this.description = description
        this.location = location
        this.transferability = transferability
        this.openTimeVisibility = openTimeVisibility
        this.closeDateVisibility = closeDateVisibility
        this.closeTimeVisibility = closeTimeVisibility
        this.cutoff = cutoff
        this.allStudentsVisibility = allStudentsVisibility
        this.bouncers = bouncers
        this.numArrived = numArrived
    }

    constructor(orgID: Int) {
        this.orgId = orgID
        this.startDate = Date().toString()
        this.endDate = Date().toString()
        this.name = ""
        this.description = ""
        this.location = ""
        this.transferability = false
        this.openTimeVisibility = false
        this.closeDateVisibility = false
        this.closeTimeVisibility = false
        this.cutoff = Date().toString()
        this.allStudentsVisibility = false
        this.bouncers = arrayOf()
        this.numArrived = 0
    }

}